extensions/pkg/terraformer/terraformer.go

```diff
@@ -297,6 +304,30 @@ func (t *terraformer) createOrUpdateTerraformerAuth(ctx context.Context) error {
 	return t.createOrUpdateRoleBinding(ctx)
 }
 
+func (t *terraformer) ensureStateHasOwnerRef(ctx context.Context) error {
+	infra := &extensionsv1alpha1.Infrastructure{}
+	if err := t.client.Get(ctx, kutils.Key(t.namespace, t.name), infra); err != nil {
+		return err
+	}
```

Comment:
I think, currently there the terraformer library doesn't know anything about the `Infrastructure` resource.
Instead it is just handed a namespace and name to use for deploying pods, but this doesn't have to be namespace/name of an infra resource. 
I'm wondering, if we should keep it like that.

One way to still add owner refs then, would be to add another function to set the owner reference on the terraformer object (extensions pass the infra object/metadata there) and propagate it from there to the initializers and so on.
If an extension doesn't call the func, no owner refs will be set.

WDYT?
I'm not sure, if there is a real usecase for the library independent from an infra reconciliation, so I don't know we need to consider this...

Reply:
To be honest, I was looking how to deliver the `ownerRef` via the context `context.WithValue(...)` and then pull the value in the `execute` function. But this seems a bit ugly for something that will be removed after several releases, thus I decided not to do it. 

The other option that I was thinking is almost the same as your idea, to add 
```golang
ownerRef         *metav1.OwnerReference
```
field in https://github.com/gardener/gardener/blob/085d163e1cbd73397aa6b31197714729f8bf7c53/extensions/pkg/terraformer/types.go#L46-L69

and initialize it `New(ForConfig)`, but now I would prefer to use setter as you propose. 
I will try it out and will come back here with the results.

Reply:
I was thinking about the same, but then refrained from mentioning it because of

> I'm not sure, if there is a real usecase for the library independent from an infra reconciliation

Probably there isn't. In a perfect world I would totally agree to not do it this way (keep the library independent of the `Infrastructure` object), but given that @vpnachev already implemented and verified, and, most importantly, this code is only temporary, I'd vote for keeping it as it is.

Reply:
I have changed the implementation with ec3693b6c059628d57964f76de8fdb90f68dc48f. 
If you are fine with it, I will squash it.